proxy/envoy/config.go

```diff
@@ -158,62 +162,88 @@ func buildSidecar(env proxy.Environment, sidecar proxy.Node) (Listeners, Cluster
-
-	listeners := append(inbound, outbound...)
-	clusters := append(inClusters, outClusters...)
-
-	// If management listener port and service port are same, bad things happen
-	// when running in kubernetes, as the probes stop responding. So, append
-	// non overlapping listeners only.
-	for i := range mgmtListeners {
-		m := mgmtListeners[i]
-		c := mgmtClusters[i]
-		l := listeners.GetByAddress(m.Address)
-		if l != nil {
-			glog.Warningf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
-				m.Name, m.Address, l.Name, l.Address)
-			continue
+	listeners := make(Listeners, 0)
+	clusters := make(Clusters, 0)
+
+	if env.Mesh.ProxyListenPort > 0 {
+		inbound, inClusters := buildInboundListeners(env.Mesh, sidecar, instances)
+		outbound, outClusters := buildOutboundListeners(env.Mesh, sidecar, instances, services, env.IstioConfigStore)
+		mgmtListeners, mgmtClusters := buildMgmtPortListeners(env.Mesh, managementPorts, sidecar.IPAddress)
+
+		listeners = append(listeners, inbound...)
+		listeners = append(listeners, outbound...)
+		clusters = append(clusters, inClusters...)
+		clusters = append(clusters, outClusters...)
+
+		// If management listener port and service port are same, bad things happen
+		// when running in kubernetes, as the probes stop responding. So, append
+		// non overlapping listeners only.
+		for i := range mgmtListeners {
+			m := mgmtListeners[i]
+			c := mgmtClusters[i]
+			l := listeners.GetByAddress(m.Address)
+			if l != nil {
+				glog.Warningf("Omitting listener for management address %s (%s) due to collision with service listener %s (%s)",
+					m.Name, m.Address, l.Name, l.Address)
+				continue
+			}
+			listeners = append(listeners, m)
+			clusters = append(clusters, c)
+		}
+
+		// set bind to port values for port redirection
+		for _, listener := range listeners {
+			listener.BindToPort = false
 		}
-		listeners = append(listeners, m)
-		clusters = append(clusters, c)
-	}
 
-	// set bind to port values for port redirection
-	for _, listener := range listeners {
-		listener.BindToPort = false
+		// add an extra listener that binds to the port that is the recipient of the iptables redirect
+		listeners = append(listeners, &Listener{
+			Name:           VirtualListenerName,
+			Address:        fmt.Sprintf("tcp://%s:%d", WildcardAddress, env.Mesh.ProxyListenPort),
+			BindToPort:     true,
+			UseOriginalDst: true,
+			Filters:        make([]*NetworkFilter, 0),
+		})
 	}
 
-	// add an extra listener that binds to the port that is the recipient of the iptables redirect
-	listeners = append(listeners, &Listener{
-		Name:           VirtualListenerName,
-		Address:        fmt.Sprintf("tcp://%s:%d", WildcardAddress, env.Mesh.ProxyListenPort),
-		BindToPort:     true,
-		UseOriginalDst: true,
-		Filters:        make([]*NetworkFilter, 0),
-	})
+	// enable HTTP PROXY port if necessary; this will add an RDS route for this port
+	if env.Mesh.ProxyHttpPort > 0 {
```

Comment:
There is no harm in having the proxy enabled on all nodes. We had 15000 with the admin enabled, we can have 15002 for proxy mode - the memory and whatever costs are minimal. If we find it is a problem - we can optimize, but adding complexity without a good reason is not a good idea.  We don't know which VM will use the proxy vs iptable, and it is possible to even have a mix on a single VM - some processes using proxy, some having to use iptable. 

Let's not make it complicated with too many options - all we need is a trivial listening on the  proxy port, and a combined route discovery, neither affects the rest of the functionality or adds risks.
Once we have it - we can test and find if there is any issue ( and how well it works ), and in case of problems revert and try something else. 

So far the tests I did with manual configuration don't seem to show any issue - can someone explain what is the 'correct endpoint port in sds clusters' - or what sds have to do with the proxy ?